packages/sanity/src/core/form/inputs/arrays/ArrayOfOptionsInput/ArrayOfObjectOptionsInput.tsx

```diff
@@ -0,0 +1,133 @@
+import React, {useCallback} from 'react'
+import {Box, Card, Checkbox, Flex, Grid} from '@sanity/ui'
+import {resolveTypeName} from '@sanity/util/content'
+import {ArraySchemaType, isKeyedObject} from '@sanity/types'
+import {set, unset} from '../../../patch'
+import {ArrayOfObjectsInputProps} from '../../../types'
+import {ChangeIndicator} from '../../../../changeIndicators'
+import {IncompatibleItemType} from '../../../members/views/IncompatibleItemType'
+
+function isEqual(item: any, otherItem: any): boolean {
+  if (item === otherItem) {
+    return true
+  }
+  if (typeof item !== typeof otherItem) {
+    return false
+  }
+  if (typeof item !== 'object' && !Array.isArray(item)) {
+    return item === otherItem
+  }
+  if (item._key && item._key === otherItem._key) {
+    return true
+  }
+  if (Array.isArray(item)) {
+    if (!item.length !== otherItem.length) {
+      return false
+    }
+    return item.every((it, i) => isEqual(item[i], otherItem[i]))
+  }
+  const keys = Object.keys(item)
+  const otherKeys = Object.keys(item)
+  if (keys.length !== otherKeys.length) {
+    return false
+  }
+  return keys.every((keyName) => isEqual(item[keyName], otherItem[keyName]))
+}
+
+function inArray(array: unknown[], candidate: unknown) {
+  return array ? array.some((item) => isEqual(item, candidate)) : false
+}
+
+function getMemberTypeOfItem(schemaType: ArraySchemaType, item: unknown) {
+  return schemaType.of.find((memberType) => memberType.name === resolveTypeName(item))
+}
+
+const EMPTY_ARRAY: unknown[] = []
+
+/**
+ * Array of predefined object options input
+ * Note: this input can handle only object values
+ */
+export function ArrayOfObjectOptionsInput(props: ArrayOfObjectsInputProps) {
```

Comment:
_type probably should be required. 

At first I thought since initialValue on array.of members dont require it, it should be the same here, but in that context it is implied, so thats why it can be added.